hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecord.java

```
@@ -553,4 +554,210 @@ protected Object getClassIdentifier() {
         return schema.getTypeName();
     }
 
+    @Nullable
+    @Override
+    public InternalGenericRecord getInternalGenericRecord(@Nonnull String fieldName) {
+        return get(fieldName, COMPACT);
+    }
+
+    @Nullable
+    @Override
+    public InternalGenericRecord[] getArrayOfInternalGenericRecord(@Nonnull String fieldName) {
+        return get(fieldName, ARRAY_OF_COMPACT);
+    }
+
+    @Nullable
+    @Override
+    public Boolean getBooleanFromArray(@Nonnull String fieldName, int index) {
+        boolean[] array = getArrayOfBoolean(fieldName);
+        if (array == null || array.length <= index) {
+            return null;
+        }
+        return array[index];
+    }
+
+    @Nullable
+    @Override
+    public Byte getInt8FromArray(@Nonnull String fieldName, int index) {
+        byte[] array = getArrayOfInt8(fieldName);
+        if (array == null || array.length <= index) {
+            return null;
+        }
+        return array[index];
+    }
+
+    @Nullable
+    @Override
+    public Character getCharFromArray(@Nonnull String fieldName, int index) {
+        char[] array = getArrayOfChar(fieldName);
```

we could throw UnsupportedOperationException here